### PR TITLE
Extract generic `HistoryView` and add it to `ModelViewSet`

### PR DESCRIPTION
**Pull request by laymonage**
Built on top of #10900 as part of #10740.

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [x] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

Check out [this branch](https://github.com/laymonage/bakerydemo/tree/country-modelviewset) of my bakerydemo fork, and from the menu, select Countries Of Origin. Open the edit view of one of the instances, and you should see the history icon link at the top right of the header that takes you to the history view. You can also filter by action/user/dates, as with the pages and snippets history view. However, the history view for a non-snippet model does not support revision handling, even if the model extends `RevisionMixin`, so no actions (e.g. "Review this version") won't be shown on the log entries. The snippets history view shall remain unchanged.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Review by laymonage — COMMENTED** [review]
OK, this should be ready to review. I tried entertaining the idea of adding `no_results_message` to the `BaseListingView` class in https://github.com/laymonage/wagtail/commit/5875b56362ca36bf7cfb118d18e4f1ebeb1f5beb and using it in https://github.com/laymonage/wagtail/commit/1618e2b23405240561fb2365beca4bc873c2209c, but then I realised I still needed the `index_results.html` template to show the "There are x results" message, so the template override was still necessary.

Will probably extract the `no_results_message` in one of the other PRs instead.

File: wagtail/admin/views/generic/history.py [hunk 1]
```diff
@@ -1,13 +1,106 @@
...
+    timestamp = django_filters.DateFromToRangeFilter(
+        label=gettext_lazy("Date"), widget=DateRangePickerWidget
+    )
+
+    class Meta:
+        model = ModelLogEntry
+        fields = ["action", "user", "timestamp"]
+
+
+class HistoryView(IndexView):
```
[the remark this answers is not in the document]
Comment (gasman) [reply wagtail/admin/views/generic/history.py:46]: Didn't we previously run in to problems with subclassing IndexView for listings that aren't really indexes (i.e. a listing of models available to perform CRUD operations on), and that prompted the creation of wagtail.admin.views.generic.BaseListingView? Looking at #10540 and the linked issues, it seems the problem was that IndexView tried to apply permission policies to the listing items. Are we likely to run into that again, and if so, is subclassing BaseListingView an option instead?

[the remark this answers is not in the document]
Comment (laymonage) [reply wagtail/admin/views/generic/history.py:46]: We can subclass `BaseListingView`, but that leaves us unable to filter the history items, as `BaseListingView` doesn't provide the filtering logic (and rightly so, as it doesn't assume it's dealing with model objects).

The history view is technically an index view for `FooLogEntry` that has been filtered to a specific model, hence subclassing `IndexView` seems to make sense in here still.

> Are we likely to run into that again

I don't think so, I don't see code in the `IndexView` that specifically does permission checks with the policy against the listing items. The checks that exist are currently only the ones for adding `can_edit` and `can_delete` context variables.

The history view, or at least the generic one (not snippets), does not have any actions on the listing, so I think we're safe.

After this PR, I'm aiming to introduce the ability to add listing buttons on the view-level, and in there I think we'll have a better idea what the issues around such views that have a separation between the "model for the main, specific object" and the "model for the objects that are actually listed".

**Review by gasman — APPROVED** [review]
Looks good to me! Nice bit of cleanup here.

**Pull request merged**